src/libraries/System.Data.Common/src/System/Data/SQLTypes/SQLGuid.cs

```diff
@@ -328,6 +316,11 @@ public static XmlQualifiedName GetXsdType(XmlSchemaSet schemaSet)
             return new XmlQualifiedName("string", XmlSchema.Namespace);
         }
 
-        public static readonly SqlGuid Null = new SqlGuid(true);
+        void ISerializable.GetObjectData(SerializationInfo info, StreamingContext context)
+        {
+            info.AddValue("m_value", ToByteArray(), typeof(byte[]));
+        }
```

Comment:
As with above, please add comments around this to explain why this is needed. I also recommend moving this method declaration up so that both sides of (de)serialization are next to each other, making the relationship more obvious.